orderbook/src/account_balances.rs

```diff
@@ -25,11 +25,19 @@ pub trait BalanceFetching: Send + Sync {
 
     // Called periodically to perform potential updates on registered balances
     async fn update(&self);
+
+    // Check if the allowance manager would be able to call transfer_from with these parameters.
+    // This is useful for tokens that are not consistent about their internal checks and what they
+    // report as balance and allowance. By checking whether the actual transfer would suceed we can
+    // be more certain (but still not 100%) that the balance really is available to the settlement
+    // contract.
+    async fn can_transfer(&self, token: H160, from: H160, amount: U256) -> bool;
```

Comment:
Not sure if BalanceFetching is the best place for this but seeing as I plan on making this part of the maintenance I think it makes sense to integrate it with the batch calls here in a future PR.

Reply:
What amount are you planning to use for the maintenance job?

Reply:
The same balance that we already fetch. Which is a little annoying because we have to fetch the balance first and then do another batch call with the transfer. Alternative we could just try `1` to catch paused tokens.

Reply:
Yeah, I think this is not super obvious. Fetching the whole allowance/balance might be too restrictive if a token e.g. implements a 0.1% transfer fee we could still sell up to 99.9% of our balance. Maybe we can add the total of all orders of that token together (similar to how we would compute the available balance a user has in a given token)? But `1` sounds like a simple and good enough solution to catch paused or frozen tokens.

Reply:
Should this also always have `from` be the allowance manager and `to` be the settlement contract address (in case it checks transfers for specific addresses? The [OpenZeppelin ERC20 implementation disallows transferring to address 0](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/3157aff844dfc19fdd68988d9c859959f2d0a4f9/contracts/token/ERC20/ERC20.sol#L213) for example).

Reply:
Yes always hardcoded.